### PR TITLE
SVC-12072 Add Qualys EUS Detection Workaround

### PR DESCRIPTION
**Pull request by bsper2**

### CODE REVIEW

**Comment by bsper2** [comment]
This is a continuation of #17 , I closed because something looked off with the rebase.

**Comment by bsper2** [comment]
This makes use of work in https://github.com/ncsa/puppet-rhsm/pull/7

File: REFERENCE.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 * [`profile_audit`](#profile_audit): A short summary of the purpose of this class
 * [`profile_audit::net_process_log`](#profile_auditnet_process_log): Logs user processes and open network connections to syslog
 * [`profile_audit::qualys`](#profile_auditqualys): Configure host to be scanned by qualys
+* [`profile_audit::qualys_eus_reporting`](#profile_auditqualys_eus_reporting): Allow qualys to sudo to root and add wrapper for subscription-manager to lie about enabled repos.
```
Comment (billglick) [inline REFERENCE.md:12]: Technically the sudo part is now separate, so this description is slightly wrong in how it is worded.

File: REFERENCE.md [hunk 2]
```diff
@@ -184,6 +200,38 @@ Data type: `String`
 
 String of the comment in passwd file of the local qualys user
 
+### <a name="profile_auditqualys_eus_reporting"></a>`profile_audit::qualys_eus_reporting`
+
+Allow qualys to sudo to root and add wrapper for subscription-manager to lie about enabled repos.
```
Comment (billglick) [inline REFERENCE.md:205]: Again description is slightly wrong in that a different option now allows the sudo part.

File: manifests/qualys_eus_reporting.pp [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# @summary Allow qualys to sudo to root and add wrapper for subscription-manager to lie about enabled repos.
```
Comment (billglick) [inline manifests/qualys_eus_reporting.pp:1]: Again this no longer allows sudo, but needs sudo/escalated scan set.

File: manifests/qualys_eus_reporting.pp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  file { '/etc/profile.d/qualys_eus_reporting.sh':
+    ensure  => $alias_ensure_parm,
+    mode    => '0644',
+    owner   => 'root',
+    group   => 'root',
+    content => file("${module_name}/qualys_eus_reporting.sh"),
+  }
+
+  # Create script that the alias will call
+  file { '/root/qualys_eus_reporting.sh':
```
Comment (billglick) [inline manifests/qualys_eus_reporting.pp:42]: Should this be in a subdirectory of `/root`, e.g. `/root/scripts`? I thought we generally have a subdirectory there for scripts like this.

**Review by billglick — APPROVED** [review]
A few minor suggestions. But acceptable as is.

File: REFERENCE.md (hunk 1 above)
Comment (bsper2) [reply REFERENCE.md]: Documentation updated

File: REFERENCE.md (hunk 2 above)
Comment (bsper2) [reply REFERENCE.md]: Documentation updated

File: manifests/qualys_eus_reporting.pp (hunk 3 above)
Comment (bsper2) [reply manifests/qualys_eus_reporting.pp]: Documentation updated

File: manifests/qualys_eus_reporting.pp (hunk 4 above)
Comment (bsper2) [reply manifests/qualys_eus_reporting.pp]: Yeah, I moved it to /root/scripts

**Pull request merged**